index.js

```diff
@@ -217,15 +230,19 @@ function adapter(uri, opts){
 
     debug('removing %s from rooms %s', id, Object.keys(rooms));
 
-    Adapter.prototype.delAll.call(this, id);
-
     if (!rooms) {
-      return process.nextTick(fn.bind(null, null));
+      if (fn) {
+        process.nextTick(fn.bind(null, null));
+      }
+      return;
     } else {
-      var self = this;
-      runCommand('unsubscribe', Object.keys(rooms).map(function(room) {
-        return self.getChannelName(room);
-      }));
+      for (var room in rooms) {
+        this.del(id, room);
```

Comment:
Oops, I found mistake.
var room is index of rooms.
We have to change this line to below.
```javascript
this.del(id, rooms[room]);
```